examples/models/resnet.py

```diff
@@ -0,0 +1,51 @@
+import torch.nn as nn
+from models.modules.layers_3d import SparseConvBlock, SparseResBlock
+
+__all__ = ['SparseResNet']
+
+
+class SparseResNet(nn.Module):
+
+    def __init__(self, in_channels: int = 4) -> None:
+        super().__init__()
+
+        cs = [16, 32, 64, 128]
+
+        self.stem = nn.Sequential(
+            SparseConvBlock(in_channels, cs[0], 3, stride=1),
+            SparseResBlock(cs[0], cs[0], 3), SparseResBlock(cs[0], cs[0], 3))
+        self.stage1 = nn.Sequential(
+            SparseConvBlock(cs[0], cs[1], 3, stride=2),
+            SparseResBlock(cs[1], cs[1], 3),
+            SparseResBlock(cs[1], cs[1], 3),
+        )
+        self.stage2 = nn.Sequential(
+            SparseConvBlock(cs[1], cs[2], 3, stride=2),
+            SparseResBlock(cs[2], cs[2], 3),
+            SparseResBlock(cs[2], cs[2], 3),
+        )
+        self.stage3 = nn.Sequential(
+            SparseConvBlock(cs[2], cs[3], 3, stride=2),
+            SparseResBlock(cs[3], cs[3], 3),
+            SparseResBlock(cs[3], cs[3], 3),
+        )
+
+        self.stage4 = SparseConvBlock(cs[3], cs[3], [1, 3, 1], stride=[1, 2, 1])
```

Comment:
I think this implementation is adopted from the detection model. I think it would be better to make this backbone a bit more general. @kentangSJTU what do you think?

Reply:
Not sure whether we need to do that this time. Perhaps in the future we can also add support for different depths for each stage.

Reply:
The general `SparseResNet` backbone has been implemented.